packages/di/src/domain/DIContext.ts

```diff
@@ -92,6 +93,13 @@ export class DIContext extends Map<any, any> implements ContextMethods {
   async emit(eventName: string, ...args: any[]) {
     return this.injector?.emit(eventName, ...args);
   }
+
+  runInContext(next: Function) {
+    return runInContext(this, () => {
+      next = this.injector?.alter("$alterRunInContext", next, this);
+      return next();
```

Comment:
This might lead to NPE since `this.injector` can be null or undefined
```suggestion
      next = this.injector?.alter("$alterRunInContext", next, this) ?? next;
      return next();
```

Reply:
well seen :)

Reply:
done  :D